actions/structs.go

```diff
@@ -13,14 +13,16 @@ type RollupInfo struct {
 	Namespace           []byte        `json:"namespace"`
 	AuthoritySEQAddress codec.Address `json:"authoritySEQAddress"`
 	SequencerPublicKey  []byte        `json:"sequencerPublicKey"`
+	StartEpoch          uint64        `json:"startEpoch"`
 }
 
-func NewRollupInfo(namespace []byte, feeRecipient codec.Address, authoritySEQAddress codec.Address, sequencerPublicKey []byte) *RollupInfo {
+func NewRollupInfo(namespace []byte, feeRecipient codec.Address, authoritySEQAddress codec.Address, sequencerPublicKey []byte, startEpoch uint64) *RollupInfo {
 	return &RollupInfo{
 		FeeRecipient:        feeRecipient,
 		Namespace:           namespace,
 		AuthoritySEQAddress: authoritySEQAddress,
 		SequencerPublicKey:  sequencerPublicKey,
+		StartEpoch:          startEpoch,
```

Comment:
it can be only upgrade if 
- startEpoch is not changing
- they are changing startEpoch to a epoch that is greater and equal than epoch n+2, which works the same as exit first then register

Reply:
does that also cover the case of modifying startEpoch when the current epoch is greater than the startEpoch? this should not happen.